docs/specs/NETWORKING.md

```diff
@@ -1,3 +1,769 @@
+It must be performed for every connection, and therefore forces nodes to 
+authenticate / prove ownership of their operator key.
+
+**TBD** Public, static nodes such as exporter requires registration 
+
+### Message Structure
+
+The following information will be exchanged as part of the handshake:
+
+```protobuf
+syntax = "proto3";
+import "gogo.proto";
+
+// AuthMessage is a message that is used for authenticating nodes
+message HandshakeMessage {
+  // info is the node information to sign
+  NodeInfo info = 1 [(gogoproto.nullable) = false];
+  // signed is a signature of the message
+  bytes signed  = 2 [(gogoproto.nullable) = false];
+
+  // TODO: TBD
+  // msg_seq helps to keep track over multiple handshakes, will by set by the sender
+  uint64 msg_seq = 1 [(gogoproto.nullable) = false];
+  // peer_id is the peer that the message is targeted for, will by set by the sender
+  bytes peer_id  = 2 [(gogoproto.nullable) = false];
+}
+
+// NodeInfo contains node's information
+message NodeInfo {
+  // peer_id of the authenticating node
+  bytes peer_id          = 1 [(gogoproto.nullable) = false];
+  // operator_id of the authenticating node
+  bytes operator_id      = 2 [(gogoproto.nullable) = true];
+  // node_type is the type of the authenticating node
+  uint64 node_type       = 3 [(gogoproto.nullable) = false];
+  // execution_node is the "name/version" of the eth1 node used by the node
+  string execution_node  = 4; // TBD: actual value
+  // consensus_node is the "name/version" of the eth2 node used by the node
+  string consesnsus_node = 5; // TBD: actual value
+  // fork_version is the current fork used by the node
+  uint32 fork_version    = 6;
+  // fork_version is the current ssv-node version
+  string node_version    = 7;
+}
+
+```
+
+---
+
+
+## Networking
+
+### Pubsub
+
+The main purpose is for broadcasting messages to a group (AKA subnet) of nodes. \
+In addition, the following are achieved as well:
+
+- subscriptions metadata helps to get liveliness information of nodes
+- pubsub scoring enables to prune bad/malicious peers based on network behavior and application-specific rules
+
+The following parameters are used for initializing pubsub:
+
+- `floodPublish` was turned on for better reliability, as peer's own messages will be propagated to a larger set of peers 
+  (see [Flood Publishing](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#flood-publishing)) 
+- `peerOutboundQueueSize` / `validationQueueSize` were increased to `600`, to avoid dropped messages on bad connectivity or slow processing
+- `directPeers` includes the exporter peer ID, to ensure it gets all messages
+- `subscriptionFilter` was injected to ensure a peer will connect to relevant topics, see [SubscriptionFilter](https://github.com/libp2p/go-libp2p-pubsub/blob/master/subscription_filter.go) interface
+- (fork `v1`) `msgID` is a custom function that calculates a `msg-id` based on the message content hash. 
```

Comment:
this needs better explanation. We basically say that each network message some deterministic msg ID which prevents re-sending the same messages time and time again?